daemon/lib/lightning/lightning.go

```diff
@@ -0,0 +1,33 @@
+package lightning
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/shopspring/decimal"
+)
+
+const DefaultCltvExpiry uint64 = 144
+
+var ErrInvoiceCanceled = fmt.Errorf("invoice canceled")
+
+// This will be the pubkey in the case of direct connection, or the dapr alias in the case of dapr
+type PubKey = string
+type Preimage = string
+type NetworkFeeSats = int64
+
+type GraphStatus struct {
+	GraphSynced bool
+	ChainSynced bool
+}
+
+//go:generate mockgen -destination=mock.go -package=lightning . Client
+type Client interface {
+	PubKey() PubKey
+	GetGraphStatus(ctx context.Context) (*GraphStatus, error)
```

Comment:
what do you envision this needed for?

Reply:
probably would be useful to be sure that the LND node is synced before initiating a swap

Reply:
And if it's not synced? What would you do? Block swaps (affecting ops) because the graph might be desync
This is not a good idea as a desync graph can be ok for doing swaps.

Ops > Routing (Fees will be constrained at other places)

Reply:
done 🏁